swift/ios-shoppe-demo/Views/StoreViewController.swift

```diff
@@ -73,9 +77,20 @@ class StoreViewController: UICollectionViewController, UICollectionViewDelegateF
 
         navigationController?.navigationBar.titleTextAttributes = textAttributes
         navigationController?.navigationBar.barTintColor = UIColor(displayP3Red: 206/255, green: 78/255, blue: 142/255, alpha: 1.0)
-
-        navigationItem.title = "iOS Shoppe"
-
+        
+        let navButton =  UIButton(type: .custom)
+        navButton.frame = CGRect(x: 0, y: 0, width: 100, height: 40)
+        navButton.setTitleColor(.white, for: .normal)
+        navButton.setTitle("iOS Shoppe", for: .normal)
+        navigationItem.titleView = navButton
+
+        navButton.rx.tap
+            .buffer(timeSpan: .milliseconds(1000), count: 4, scheduler: MainScheduler.instance)
+            .filter{$0.count >= 4}
+            .subscribe(onNext: { value in
+                fatalError("Clicked On View 4 Times, Will Crash Now, Good Job!")
```

Comment:
Is `fatalError` a crash? The [documentation](https://developer.apple.com/documentation/swift/1538698-fatalerror) only promises that it "stops execution" which sounds more like an `exit(0)` than a segfault or exception. I'm not sure if there is a way to handle/catch it. `fatalError` is also mysteriously Swift-only with [no exact equivalent in ObjC](https://stackoverflow.com/questions/37266631/is-there-an-objective-c-equivalent-to-swifts-fatalerror).

Reply:
Cool call @martin-fs , and I don't have a good answer. I need to read up on this, thanks for the callout.

Reply:
I think `fatalError` works for the purposes of this PR, but in the interest of future proofing this a bit, I'm going to `raise` an `NSException` since that is likely where we will start when we add basic crash reporting on iOS. Though, you cannot catch `NSException` in Swift as they're not compatible (they're compatible with Coacoa `NSError`). 

I'll change to something like: 

```
NSException(name: "ClickCrash", reason: "Clicked On View 4 Times, Will Crash Now, Good Job!", userInfo: nil).raise()
```